src/transformers/modeling_utils.py

```diff
@@ -1150,8 +1271,23 @@ def _generate_beam_search(
             assert (len(hypo) == max_length for hypo in best)
             decoded = torch.stack(best).type(torch.long).to(next(self.parameters()).device)
 
+        if self.config.is_encoder_decoder:
+            # do not return first <EOS> token
+            return decoded[:, 1:]
         return decoded
 
+    # force one of token_ids to be generated by setting prob of all other tokens to 0.
+    def _force_token_ids_generation(self, scores, token_ids):
```

Comment:
As mentioned above in `_prepare_scores...`, I don't understand why we need this.

Reply:
see my comment in `_prepare_scores`